File: snippets/contact-center/create-enqueue_call-endpoint/create-enqueue_call-endpoint.cs
public ActionResult enqueue_call()
{
    var response = new VoiceResponse();
            
    Dictionary<string, string> department = new Dictionary<string, string>();
    department.Add("1", "sales");
    department.Add("2", "support");
    department.Add("3", "billing");

    var enqueue = new Enqueue(workflowSid: workflow_sid]);

    enqueue.Task("{'selected_product:'@" + department[Request.Params.Get("digits")] + @"'}");
    response.Append(enqueue);
            
    return Content(response.ToString(), contentType: "text/xml");
         
}

File: snippets/contact-center/create-enqueue_call-endpoint/create-enqueue_call-endpoint.py
@app.route("/enqueue_call", methods=["GET", "POST"])
def enqueue_call():
     if 'Digits' in request.values:
        choice = int(request.values['Digits'])
        dept = {
            1: "sales",
            2: "support",
            3: "billing"
        }
        resp = VoiceResponse()
        resp.say('Thank you, connecting you now')
        with resp.enqueue(workflow_sid=workflow_sid) as e:
            e.task('{"selected_product" : "' + dept[choice] + '"}')
        return Response(str(resp), mimetype='text/xml')
    else:
        resp = VoiceResponse()
        resp.say("no digits detected")   #tell user something is amiss
        resp.redirect("/incoming_call")  #redirect back to initial step
    return Response(str(resp), mimetype='text/xml')
